sdk/core/core-http/lib/policies/deserializationPolicy.ts

```diff
@@ -153,6 +153,20 @@ export function deserializeResponseBody(jsonContentTypes: string[], xmlContentTy
                       : [];
                   }
                   error.body = operationSpec.serializer.deserialize(defaultResponseBodyMapper, valueToDeserialize, "error.body");
+                  error.parsedBody = error.body;
+                }
+              }
+
+              if (parsedResponse.headers) {
+                const defaultResponseHeadersMapper: Mapper | undefined =
```

Comment:
no need to assign a type to the variable, since it can be inferred from its initial value.

Reply:
I would suggest to collapse the 2 if conditions and skip the constant

```javascript
if (parsedResponse.headers && defaultResponseSpec.headersMapper) {
   error.parsedHeaders =operationSpec.serializer.deserialize(...);
}
```

Reply:
Done